test/common/upstream/cluster_manager_impl_test.cc

```diff
@@ -260,6 +260,46 @@ TEST_F(ClusterManagerImplTest, UnknownHcType) {
   EXPECT_THROW(create(*loader), EnvoyException);
 }
 
+TEST_F(ClusterManagerImplTest, MaxClusterName) {
+  std::string json = R"EOF(
+  {
+    "clusters": [
+    {
+      "name": "clusterwithareallyreallylongnamemorethanmaxcharsallowedbyschema"
+    }]
+  }
+  )EOF";
+
+  Json::ObjectPtr loader = Json::Factory::LoadFromString(json);
+  try {
+    create(*loader);
+  } catch (Json::Exception& e) {
+    EXPECT_EQ("JSON object doesn't conform to schema.\n Invalid schema: #/properties/name.\n "
```

Comment:
"I disagree, this is a better test." yup, we quickly chatted with Jose on this offline and he was expressing similar concerns, which is a valid point.